### ISSUE TITLE
After using MKL native provider, Process.Start() hangs

### ISSUE THREAD

**Issue by vrcmh** [mathnet/mathnet-numerics#583]
Consider this small test program:

```
Control.UseNativeMKL();
var M1 = Matrix<double>.Build.DenseIdentity(100);
var M2 = Matrix<double>.Build.DenseIdentity(100);
var M3 = M1.QR().Solve(M2);
Console.WriteLine("Starting process");
Process.Start("/Applications/Utilities/Terminal.app");
```

Process.Start hangs indefinitely, mono is using 100% CPU. It does not hang if I use the managed implementation (remove the line Control.UseNativeMKL()). It also does not hang when using smaller matrices (e.g. 10x10).

Environment:
Mac OS X 10.12.6
Math.net Numerics 4.5.1
Intel MKL 2018

What could be the problem?

**Comment by cdrnet** [comment]
Have you tried setting `UseShellExecute` to `false`, in `ProcessStartInfo`?

**Comment by vrcmh** [comment]
Yes, same behavior with UseShellExecute=false, and the same thing also
happens if you p/invoke execve directly or use
Mono.Unix.Native.Syscall.execve.



Christoph Ruegg <notifications@github.com> schrieb am Mo., 9. Juli 2018,
04:43:
